staging/src/k8s.io/apiextensions-apiserver/pkg/registry/customresource/strategy.go

```diff
@@ -114,6 +115,13 @@ func (a customResourceStrategy) PrepareForCreate(ctx context.Context, obj runtim
 	}
 
 	accessor, _ := meta.Accessor(obj)
+	if len(accessor.GetAnnotations()[genericapirequest.AnnotationKey]) > 0 {
+		// in general the shard annotation is not attached to objects, instead, it is assigned by the storage layer on the fly
+		// to avoid an additional UPDATE request (mismatch on the generation field) replicated objects have the shard annotation set
+		// thus we need to remove the shard annotation and simply return early so that the generation is not reset to 1
+		delete(accessor.GetAnnotations(), genericapirequest.AnnotationKey)
```

Comment:
don't think this is correct (have seen the old thread). `GetAnnotations()` returns a `map[string]string`. In unstructured, it is  `map[string]interface{}`. Hence, there must be a deepcopy with type cast happening.

Reply:
To avoid the deepcopy use the `*NoCopy` unstructured helpers.

Reply:
brought back the previous version, I don't think we need to use `NoCopy`. Using `GetAnnotaions/SetAnnotations` is always correct.

Reply:
`serverResource` uses `unstructuredCreator`, that means requests to that EP will operate on `unstructured data` https://github.com/kcp-dev/kubernetes/blob/3ddc03bc8d00fb495d03c4014d09809f4d441f3f/staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/customresource_handler.go#L871